### PR TITLE
Tech 15720 add buildkite and upgrade ruby

### PR DESCRIPTION
**Pull request by brandonbell**
### Summary
Add `buildkite` pipeline.
Update base image to Ruby 2.7.8 Bullseye 
Update Ruby Version to 2.7.8

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+- label: ":ruby: Unit Tests"
+  timeout_in_minutes: 30
+  notify:
+    - github_commit_status:
+        context: clean-build
+  agents:
+    queue: ruby-2-7
+  command:
+    - bundle install
+    - bundle exec rspec
```
Comment (praymann) [inline .buildkite/pipeline.yml:12]: ❓ Looks like TravisCI was doing `bundle exec rake test`, was there a reason to jump around the Rakefile?

**Review by praymann — APPROVED** [review]
Looks good, left a comment since I'm curious about the move away from the `Rakefile` (if we're not going to use it then we should remove it, and the dependency on `rake`).

Side note, looks like the build goes green but the Coverage through `coveralls` blows up.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (praymann) [reply .buildkite/pipeline.yml:12]: Speaking of TravisCI, might as well knife that file?

Comment (brandonbell) [reply .buildkite/pipeline.yml:12]: No reason, just me being less inclined to use `rake` out of habit.  I'll look into `coveralls` too, not familiar with that.  

Thanks for the feedback!

**Pull request merged**